_posts/2023-11-15-Codesgiving-Part-1.md

```diff
@@ -0,0 +1,289 @@
+
+Use [timeboxing techniques](https://cu-dbmi.github.io/set-website/2023/01/17/Timebox-Your-Software-Work.html) to  help make sure the time you spend in development is no more than necessary.
+Also recall that a "simplest version" of the code can often be more quickly refactored and completed than devising a "perfect" solution the first time.
+Remember, you'll also very likely have the help of a code review before the code is merged (expect to learn more and add changes during review!).
+
+## Code Review, Revisions, and Post-actions
+
+### Pull Requests and Code Review
+
+When your code and new test(s) are in a good spot it's time to ask for a code review.
+Be sure to follow the open-source [`CONTRIBUTING.md` document](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) documentation related to the project; each one can vary.
+When working on GitHub-based projects, you'll need to open a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests) on the correct branch (usually upstream `main`).
+If you used a GitHub issue to help report the issue, mention the issue in the pull request description using the `#<issue number` reference to help link the work to the reported need (this will cause the pull request to show up within the issue).
+
+Note: It might feel tempting to perfect the code.
+Instead, consider whether the code is "good enough" and would benefit from someone else providing feedback.
+Code review takes advantage of a strength of our species: collaborative & multi-perspectival thinking.
+Leverage this in your open-source experience by seeking feedback when things feel "good enough".
+
+### Code Revisions
+
+> "Perfection is achieved, not when there is nothing more to add, but when there is nothing left to take away."
+> \- Antoine de Saint-Exupery
+
+You may be asked to update your code based on automated code quality checks or reviewer request.
+Treat these with care; embrace learning and remember that this step can take 25% of the total time for the contribution.
+When working on GitHub forks or branches, you can make additional commits directly on the development branch which was used for the pull request.
+If your reviewers requested changes, re-request their review once changes have been made to help let them know the code is ready for another look.
+
+### Post-actions and Tidying Up Afterwards
+
+Once the code has been accepted by the reviewers and through potential automated testing suite(s) the content is ready to be merged.
+Oftentimes this work is completed by core maintainers of the project.
+After the code is merged, be sure to take time to clean up your workspace by deleting your development branch and syncing with the upstream repository (optional).
```

Comment:
"it's usually a good idea to clean up..." rather than "(optional)"?

Reply:
Thanks, that sounds much better - I'll work this in!